### ISSUE TITLE
Albums: Set a default sort order for manually managed albums, moments, months and folders

### ISSUE THREAD

**Issue by Bart1909** [photoprism/photoprism#405]
Is it possible to set a default order, so that every album / folder will get these order automatically?
At the moment it is "last added" and I want to change it to "oldest first"

**Comment by lastzero** [comment]
Default for manually created albums should already be **oldest first**. The current sort order will be saved in album metadata, so that all users see the same. So when you change it, it won't revert back to whatever the default was. There can be different defaults for moments, calendar months and other automatically created albums.

**Comment by Bart1909** [comment]
I'm currently only using the "folders" and not the "albums".
There the sort is actually last added.

Is there a difference between albums and folders?

**Comment by lastzero** [comment]
Right, folders show new files first and you might want to check if there is something new in a folder since you checked last time. You can "clone" folders using the clipboard (select and click bottom right) to create regular albums from them or you simply select all contents (might be a lot though).

**Comment by Bart1909** [comment]
Okay. Is there a way to automatically create an album for new added folders?

**Comment by lastzero** [comment]
They should automatically show up in "folders" after indexing.

**Comment by Bart1909** [comment]
Correct, but then the default sort order is "new files first".
The question is, if there is a way to automatically create an album for this folders (so that the order will be "oldest first") without "cloning" the folder into an album

**Comment by lastzero** [comment]
Regular albums will never be created automatically. That makes them different from folders, moments and calendar months.

It would be possible to add an "albums" section to Settings, so that default sort orders can be changed by the user.

**Comment by diegantobass** [comment]
Hi @lastzero, thanks for photoprism, it really hits a sweet spot between ease of use and functionnality. Great project.

My situation is similar to @Bart1909 : I have ~400 folders and no albums. The sort order of the folders doesn't suit my need. I would like to automatically create an album for each folder, so that I can browse and order my pictures from albums, as I think was intended. In your answer to Bart you said :

> Edit: As a developer, you can run an update statement on the albums table and change all sort orders to whatever you prefer. More UI config options and batch operations will be added after our release.

Could you please develop this? Do you mean running update commands on the backend database? If so would you please tell us how to do this? I don't think there is a UI option or a docker command for automatic creation of albums at the moment. Do you plan on working on this? Where would someone willing to contribute would start?

thanks in advance,
cheers

**Comment by graciousgrey** [comment]
> Could you please develop this?

That's why we have this ticket ;)

>Do you mean running update commands on the backend database? If so would you please tell us how to do this?

In the database you find the table "albums"  if you want to change the default sort order for folders you need to set "album_order" to the value of your choice (e.g. "newest", "oldest", "added", "name") for all entries with "album_type"="folder"

> I don't think there is a UI option or a docker command for automatic creation of albums at the moment. Do you plan on working on this?

Can you tell us about the reason you want to convert all folders to albums instead of browsing them in the Folders section? If it is only because of the fact that the sort order is not configurable yet via the UI - we prefer to improve the folder section instead ;) If there's another reason we can consider this of course!

**Comment by graciousgrey** [comment]
Related: https://github.com/photoprism/photoprism/issues/353

**Comment by diegantobass** [comment]
> In the database you find the table "albums" if you want to change the default sort order for folders you need to set "album_order" to the value of your choice (e.g. "newest", "oldest", "added", "name") for all entries with "album_type"="folder"
> 
Thanks for this workaround !

> > I don't think there is a UI option or a docker command for automatic creation of albums at the moment. Do you plan on working on this?
> 
> Can you tell us about the reason you want to convert all folders to albums instead of browsing them in the Folders section? If it is only because of the fact that the sort order is not configurable yet via the UI - we prefer to improve the folder section instead ;) If there's another reason we can consider this of course!
>
That's my main reason and I agree with you on the fact that it is preferable to work on the UI of the folder section. So glad to see that it is on the roadmap (issue #353) !

Just to be clear, automatic or assisted creation of albums is an interesting feature per se and that could be expanded. I'm sure you have thought about it and please feel free to ignore my suggestions, but building up on the already implemented possibility of creating an album based on a multiple selection of folders and/or pictures, you could either refine the search mecanism to include the possibility of searching for specific exif values, and then create albums based on this. Or, closer to what I was suggesting, you could have a mecanism where you create albums based on a disjonction of values : using groupby, you could have an album created for each "Camera" field of exit data, or "Lens" field whatever.

Sorry I'm thinking aloud here. Thanks for the great work. Photoprism rocks.

**Comment by graciousgrey** [comment]
> Just to be clear, automatic or assisted creation of albums is an interesting feature per se and that could be expanded. I'm sure you have thought about it and please feel free to ignore my suggestions, but building up on the already implemented possibility of creating an album based on a multiple selection of folders and/or pictures, you could either refine the search mecanism to include the possibility of searching for specific exif values, and then create albums based on this. Or, closer to what I was suggesting, you could have a mecanism where you create albums based on a disjonction of values : using groupby, you could have an album created for each "Camera" field of exit data, or "Lens" field whatever.

Conditional albums are planned: https://github.com/photoprism/photoprism/issues/1431 :)

**Comment by Rathna-K** [comment]
Can you sort by folder name as the names of my folders have year in them and would be nice to sort the new ones up top.

**Comment by lastzero** [comment]
Have you tested our development preview and/or public demo yet? https://demo.photoprism.app/library/folders

- https://github.com/photoprism/photoprism/issues/353#issuecomment-1370956482

**Comment by Rathna-K** [comment]
Thanks, in the demo, I see Name in the sort options but how can one sort by Name in Asc/ descending? For example so 2023 comes as first folder instead of the first folder it scanned in Ascending order. Since the library is being built from scratch, photoprism seems to be scanning the folders in ascending order and showing them in the same order when sorted by name. Anyways let me play around with the demo a bit and see if I missed anything. Thank you.

**Comment by lastzero** [comment]
Have you tried the oldest first setting already?

**Comment by Rathna-K** [comment]
how do I enabled that down arrow that I see in the demo in my own instance? is there a config option that I should be adding to the compose file? right now in Folder view I dont see the arrow so no sorting options at all.

**Comment by lastzero** [comment]
Upgrade to our Development Preview as described in our docs.

**Comment by Rathna-K** [comment]
got it, I see it now. thanks! oh and I don't need a full-rescan right? just went thru a week and it still not complete as it is!

**Comment by lastzero** [comment]
No, no rescan required for album sorting as it uses existing metadata.

**Comment by huber75** [comment]
Hi,

not quite sure, whether I understand the sorting options of **folders** correct:

Have 270 folders which - for now - are sorted by name; hence the oldest one 2004-05-13 - xxx is on top, 2023-07-15 - xxx at the bottom. Do I have a chance to sort them by name in reverse order: newest on top? "Newest first" by "Favorites" does not bring this result

**Comment by lastzero** [comment]
While it was not possible to get a UI ready for release in the time available, these changes should allow you to manually set a default sort order for each album type in your [`config/settings.yml`](https://docs.photoprism.app/getting-started/config-files/settings/) file, e.g:

```yaml
Albums:
  Order:
    Album: oldest
    Folder: added
    Moment: oldest
    State: newest
    Month: oldest
```

In a future release, we may make these settings available to admins and users through the web interface - if there is enough demand for it? 🤔
